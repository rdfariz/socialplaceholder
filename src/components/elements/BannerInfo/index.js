import React from 'react';
import PropTypes from 'prop-types';

function Component (props) {
  const { label, title, subtitle, className } = props;
  
  return (
    <div className={`w-full p-5 rounded-lg border ${className}`}>
      {label && <p className="text-sm">{label}</p>}
      {title && <h2 className="text-lg font-bold">{title}</h2>}
      {subtitle && <p className="text-sm">{subtitle}</p>}
    </div>
  );
};

Component.defaultProps = {
  label: '',
  title: '',
  subtitle: '',
  className: ''
};

Component.propTypes = {
  label: PropTypes.string,
  title: PropTypes.string,
  subtitle: PropTypes.string,
  className: PropTypes.string 
};

export default Component;
