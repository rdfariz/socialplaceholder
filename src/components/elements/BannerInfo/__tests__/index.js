import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import BannerInfo from '../';

describe('src/components/elements/BannerInfo', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(
      <BannerInfo label="test" title="title" subtitle="test subtitle" />
    );
    expect(tree).toMatchSnapshot();
  });
});
