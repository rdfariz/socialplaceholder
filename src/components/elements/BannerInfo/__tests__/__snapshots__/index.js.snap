// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/BannerInfo render 1`] = `
<div
  className="w-full p-5 rounded-lg border "
>
  <p
    className="text-sm"
  >
    test
  </p>
  <h2
    className="text-lg font-bold"
  >
    title
  </h2>
  <p
    className="text-sm"
  >
    test subtitle
  </p>
</div>
`;
